### PR TITLE
Replace `ace` with `monaco`

### LINKED ISSUE #5391: Replace text editors: `ace` with `monaco`
**Summary**
Replace the `ace` editor with `monaco` in both the deployment and pkg details views.

**Background and rationale**
In https://github.com/vmware-tanzu/kubeapps/issues/4396 we are trying to replace the "basic forms" support with a tabular view. Along with this change, the "diff" view is ought to be modified as well for better overall performance. `Monaco` (the underlying editor used in vs code) seems to work pretty decently as it spins webworkers under the hood, leaving the main loop out of this expensive tasks (computing diffs, etc).

As a consequence, having two editors becomes unnecessary and we can safely replace the `ace` editor with `monaco`.

**Description**
Replace `ace` with `monaco` in every place that the former was being used.

**Acceptance criteria**
No `ace` editors are used.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by antgamdia**
### Description of the change

As described in #5391, this PR (required for #4396) is replacing `ace` with `monaco`. 

### Benefits

The codebase will be ready to use monaco as the default text editor.

### Possible drawbacks

N/A

### Applicable issues
- fixes #5391 
- related #4396

### Additional information

For OSM-reporting internal purposes, I'm squeezing a couple of deps I will need in subsequent PRs.

Tests are currently failing due to ESM/UMD related issues; I'm working on it.

#### Pkg details view

image

#### Installation/upgrade view

image

#### Installation/upgrade diff view (temporary, it will be changed in the next PRs)

image

### CODE REVIEW

**Comment by antgamdia** [comment]
We're finally getting rid of `ace` 🎉  see  https://github.com/vmware-tanzu/kubeapps/issues/3480#issue-1004407009, https://github.com/vmware-tanzu/kubeapps/pull/2528#pullrequestreview-608930161, https://github.com/vmware-tanzu/kubeapps/pull/2534#pullrequestreview-609965017

**Comment by antgamdia** [comment]
The building is failing due to a memory issue:

```` 
FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
`````

I guess we need to increase the mem for this CI step :S

File: dashboard/package.json [hunk 1]
```diff
@@ -32,6 +32,8 @@
     "@clr/ui": "^13.8.2",
     "@improbable-eng/grpc-web": "^0.15.0",
     "@paciolan/remote-component": "^2.12.1",
+    "@tanstack/match-sorter-utils": "^8.1.1",
+    "@tanstack/react-table": "^8.5.13",
```
Comment (antgamdia) [inline dashboard/package.json:36]: These two deps will be required in upcoming PRs

File: dashboard/src/components/AppUpgrade/AppUpgrade.test.tsx [hunk 2]
```diff
@@ -115,6 +115,41 @@ beforeEach(() => {
   spyOnUseHistory = jest
     .spyOn(ReactRouter, "useHistory")
     .mockReturnValue({ push: jest.fn() } as any);
+  // mock the window.matchMedia for selecting the theme
```
Comment (antgamdia) [inline dashboard/src/components/AppUpgrade/AppUpgrade.test.tsx:118]: We need to mock `window.matchMedia`, `window.ResizeObserver` and `HTMLCanvasElement.getContext()` as they are being used by monaco editor under the hood. The `jsdom` that `jest` uses does not include them, unfortunately.

File: dashboard/craco.config.js [hunk 3]
```diff
@@ -11,11 +12,15 @@ module.exports = {
...
         }),
         new webpack.ProvidePlugin({
-          process: 'process/browser.js',
-          Buffer: ['buffer', 'Buffer'],
+          process: "process/browser.js",
+          Buffer: ["buffer", "Buffer"],
+        }),
+        new MonacoWebpackPlugin({
+          // see https://github.com/microsoft/monaco-editor/tree/main/webpack-plugin
+          languages: ["yaml"],
```
Comment (antgamdia) [inline dashboard/craco.config.js:23]: We are bundling monaco-editor with a single language, as it is the only one being used currently. It helps decrease the webpack build mem footprint.

File: .circleci/config.yml [hunk 4]
```diff
@@ -722,12 +722,18 @@ jobs:
...
+    resource_class: large
     docker:
       - image: cimg/go:<< pipeline.parameters.GOLANG_VERSION >>
         auth:
           username: $DOCKER_USERNAME
           password: $DOCKER_PASSWORD
+    # Note that the max old space setting is per worker.
+    # Webpack uses "terser-webpack-plugin", which uses #CPUs-1 workers by default.
+    # We are using 3 workers, so we need to set the max old space to < 4GB/3, so
+    # setting it to 1GB.
```
Comment (antgamdia) [inline .circleci/config.yml:734]: More info at https://discuss.circleci.com/t/debugging-random-node-js-build-failures/39667
The `react-create-application` dep comes with plenty of new dark magic I discover every day 😅

**Review by ppbaena — APPROVED** [review]

**Pull request merged**